File: LATIHAN_ARRAY/ts/c2.ts
type Informasi_elemen = {
    value: any;
    index: number;
    type: string;
};

type informasi_elemen = (arr: any[]) => Informasi_elemen[];

const analyzeElements: informasi_elemen = (arr) => {
    const hasil: Informasi_elemen[] = []; // optional: kasih tipe eksplisit

    for (let i = 0; i < arr.length; i++) {
        hasil.push({
            value: arr[i],
            index: i,
            type: typeof arr[i],
        });
    }

    return hasil;
};

File: LATIHAN_ARRAY/rs/src/c_array/c2.rs
#[derive(Debug, Clone)]
pub enum Value {
    Int(i32),
    Bool(bool),
    Str(String),
}

#[derive(Debug, Clone)]
pub struct InformasiElemen {
    pub value: Value,
    pub index: usize,
    pub r#type: String, // gunakan r#type karena 'type' keyword Rust
}

pub fn analyze_elements(arr: &[Value]) -> Vec<InformasiElemen> {
    arr.iter()
        .enumerate()
        .map(|(i, val)| InformasiElemen {
            value: val.clone(),
            index: i,
            r#type: match val {
                Value::Int(_) => "number".to_string(),
                Value::Str(_) => "string".to_string(),
                Value::Bool(_) => "boolean".to_string(),
            },
        })
        .collect()
}
